src/fmu/sumo/sim2sumo/common.py

```diff
@@ -93,218 +91,135 @@ def filter_options(submod, kwargs):
     return filtered
 
 
-def find_full_path(datafile, paths):
-    """Find full path for datafile from dictionary
+def find_datafiles(seedpoint=None):
+    """Find datafiles relative to an optional seedpoint or the current working directory.
 
     Args:
-        datafile (str): path or name of path
-        paths (dict): dictionary of file paths
+        seedpoint (str|Path|list, optional): Specific file, list of directories, or single directory to search for datafiles.
 
     Returns:
-        Path: path to the full datafile
+        list: The datafiles found with unique stem names, as full paths.
     """
-    logger = logging.getLogger(__file__ + ".find_full_path")
-    data_name = give_name(datafile)
-    try:
-        return paths[data_name]
-    except KeyError:
-        logger.warning(
-            "Datafile %s, with derived name %s, not found in %s, have to skip",
-            datafile,
-            data_name,
-            paths,
-        )
-        return None
-
-
-def find_datafile_paths():
-    """Find all simulator paths
-
-    Returns:
-        dict: key is name to use in sumo, value full path to file
-    """
-    logger = logging.getLogger(__file__ + ".find_datafile_paths")
-    paths = {}
-    for data_path in find_datafiles_no_seedpoint():
-        name = give_name(data_path)
+    logger = logging.getLogger(__file__ + ".find_datafiles")
+    valid_filetypes = [".DATA", ".afi", ".in"]
+    datafiles = []
+    cwd = Path().cwd()  # Get the current working directory
 
-        if name not in paths:
-            paths[name] = data_path
+    if isinstance(seedpoint, dict):
+        # Extract the values (paths) from the dictionary and treat them as a list
+        seedpoint = list(seedpoint.values())
+    elif isinstance(seedpoint, list):
+        # If seedpoint is a list, ensure all elements are strings or Path objects
+        seedpoint = [Path(sp) for sp in seedpoint]
+    else:
+        seedpoint = [seedpoint]
+    if seedpoint:
+        for sp in seedpoint:
+            full_path = (
+                cwd / sp if not sp.is_absolute() else sp
+            )  # Make the path absolute
```

Comment:
This crashes if `seedpoint=None` with `NoneType object has no attribute 'is_absolute'`
Should be fixed if the `else` is changed to `elif seedpoint is not None` or something

Reply:
Should be resolved in cf3c85b.